src/algebra/module/zlattice.lean

```diff
@@ -0,0 +1,208 @@
+
+/-- The map that sends a vector `E` to the fundamental domain of the lattice,
+see `zspan.fract_mem_fundamental_domain`. -/
+def zspan.fract_map : E → E := λ m, m - zspan.floor_map b m
+
+@[simp]
+lemma zspan.fract_map_single (m : E) (i : ι):
+  b.repr (zspan.fract_map b m) i = int.fract (b.repr m i) :=
+by rw [zspan.fract_map, map_sub, finsupp.coe_sub, pi.sub_apply, zspan.floor_map_single, int.fract]
+
+@[simp]
+lemma zspan.fract_map_zspan_add (m : E) (v : E) (h : v ∈ span ℤ (set.range b)) :
+  zspan.fract_map b (v + m) = zspan.fract_map b m :=
+begin
+  refine (basis.ext_elem_iff b).mpr (λ i, _),
+  simp_rw [zspan.fract_map_single, int.fract_eq_fract],
+  use (zspan.basis b).repr ⟨v, h⟩ i,
+  simp only [map_add, finsupp.coe_add, pi.add_apply, add_tsub_cancel_right, zspan.repr_apply,
+    coe_mk],
+end
+
+lemma zspan.mem_fundamental_domain {x : E} :
+  x ∈ (zspan.fundamental_domain b) ↔ zspan.fract_map b x = x :=
+by simp only [basis.ext_elem_iff b, zspan.fundamental_domain, set.mem_Ico, set.mem_set_of_eq,
+  zspan.fract_map_single, int.fract_eq_self]
+
+lemma zspan.fract_map_mem_fundamental_domain (x : E) :
+  zspan.fract_map b x ∈ zspan.fundamental_domain b :=
+by simp only [zspan.mem_fundamental_domain, basis.ext_elem_iff b, zspan.fract_map_single,
+  int.fract_fract, eq_self_iff_true, implies_true_iff]
+
+lemma zspan.fract_map_eq_iff (m n : E) :
+zspan.fract_map b m = zspan.fract_map b n ↔ m - n ∈ span ℤ (set.range b) :=
+begin
+  rw basis.ext_elem_iff b,
+  simp only [int.fract_eq_fract, zspan.mem_span_iff, zspan.fract_map_single, linear_equiv.map_sub,
+  finsupp.coe_sub, pi.sub_apply],
+end
+
+lemma zspan.fract_map_le (m : E) :
+  ‖zspan.fract_map b m‖ ≤ finset.univ.sum (λ j, ‖b j‖) :=
+begin
+  calc
+    ‖zspan.fract_map b m‖
+        = ‖finset.univ.sum (λ i, b.repr (zspan.fract_map b m) i • b i)‖ : by rw b.sum_repr
+    ... = ‖finset.univ.sum (λ i, int.fract (b.repr m i) • b i)‖ : by simp_rw zspan.fract_map_single
+    ... ≤ finset.univ.sum (λ i, ‖int.fract (b.repr m i) • b i‖) : norm_sum_le _ _
+    ... ≤ finset.univ.sum (λ i, ‖int.fract (b.repr m i)‖ * ‖b i‖) : by simp_rw norm_smul
+    ... ≤ finset.univ.sum (λ j, ‖b j‖) : finset.sum_le_sum _,
+    intros i _,
+    rw [real.norm_eq_abs, int.abs_fract],
+    refine le_trans (mul_le_mul_of_nonneg_right (le_of_lt (int.fract_lt_one _)) (norm_nonneg _)) _,
+    rw one_mul,
+end
+
+end fintype
+
+lemma zspan.metric.bounded_fundamental_domain [finite ι] :
+  metric.bounded (zspan.fundamental_domain b) :=
+begin
+  casesI nonempty_fintype ι,
+  use 2 * finset.univ.sum (λ j, ‖b j‖),
+  intros x hx y hy,
+  refine le_trans (dist_le_norm_add_norm x y) _,
+  rw [← (zspan.mem_fundamental_domain b).mp hx, ← (zspan.mem_fundamental_domain b).mp hy],
+  refine (add_le_add (zspan.fract_map_le b x) (zspan.fract_map_le b y)).trans _,
+  linarith,
+end
+
+lemma zspan.measurable.fundamental_domain [measurable_space E] [opens_measurable_space E]
```

Comment:
I think this naming isn't right - maybe something like `zspan.fundamental_domain_measurable` is the correct naming? similarly above

Reply:
Yeah, you are probably right. I thought I'd name it using the same order of words as in the statement, but it does sound strange...